redbot/cogs/audio/apis/interface.py

```diff
@@ -640,16 +587,15 @@ async def spotify_enqueue(
 
                     if not player.current:
                         await player.play()
-            if len(track_list) == 0:
-                if not has_not_allowed:
-                    raise SpotifyFetchError(
-                        message=_(
-                            "Nothing found.\nThe YouTube API key may be invalid "
-                            "or you may be rate limited on YouTube's search service.\n"
-                            "Check the YouTube API key again and follow the instructions "
-                            "at `{prefix}audioset youtubeapi`."
-                        ).format(prefix=ctx.prefix)
-                    )
+            if not track_list and not has_not_allowed:
+                raise SpotifyFetchError(
+                    message=_(
+                        "Nothing found.\nThe YouTube API key may be invalid "
+                        "or you may be rate limited on YouTube's search service.\n"
+                        "Check the YouTube API key again and follow the instructions "
+                        "at `{prefix}audioset youtubeapi`."
+                    ).format(prefix=ctx.prefix)
```

Comment:
THIS IS PART 2/8 THAT NEED TO BE APPLIED TOGETHER WITH THIS.

```suggestion
                    )
```